docs/source/examples/ddd_tutorial.myst

```diff
@@ -0,0 +1,250 @@
+# Clifford depths.
+depths = [1, 5, 9]
+```
+
+We also define a graph representation of our qubits and assume a line topology.
+
+```{code-cell} ipython3
+# Assume chain-like connectivity
+topology = nx.Graph()
+topology.add_edges_from([(0, 1), (1, 2)])
+nx.draw(topology, with_labels=True)
+
+# Add reversed edges to topology graph.
+# This is important to represent CNOT gates with target and control reversed.
+topology = nx.to_directed(topology)
+```
+
+## Define the circuit
+
+We use mirror circuits to benchmark the performance of the device.
+Mirror circuits, introduced in *Proctor et al. (2021)* {cite}`Proctor_2021_NatPhys`, are designed such that only one bitstring
+should be sampled. 
+When run on a device, any other measured bitstrings are due to noise.
+The frequency of the correct bitstring is our target metric.
+
+```{note}
+Mirror circuits build on Loschmidt echo circuits - i.e., circuits of the form $U U^\dagger$ for some unitary $U$.
+Loschmidt echo circuits are good benchmarks but have shortcomings - e.g., they are unable to detect coherent errors.
+Mirror circuits add new features to account for these shortcomings. 
+For more background, see [arXiv:2008.11294](https://arxiv.org/abs/2008.11294).
+```
+
+To define a mirror circuit, we need the device graph. 
+We will use a subgraph of the device, and our first step is picking a subgraph with good qubits.
+
+
+### Generate mirror circuit
+
+Now that we have the device graph, we can generate a mirror circuit and the bitstring it should sample as follows.
+
+```{code-cell} ipython3
+def get_circuit(depth: int, seed: int) -> Tuple[cirq.Circuit, str]:
+    circuit, correct_bitstring = benchmarks.generate_mirror_circuit(
+        nlayers=depth,
+        two_qubit_gate_prob=1.0,
+        connectivity_graph=topology,
+        two_qubit_gate_name="CNOT",
+        seed=seed,
+        return_type="cirq",
+    )
+    return circuit, correct_bitstring
+```
+
+## Define the executor
+
+Now that we have a circuit, we define the `execute` function which inputs a circuit and returns an expectation value - here, the
+frequency of sampling the correct bitstring.
+
+```{code-cell} ipython3
+def execute(
+    circuit: cirq.Circuit,
+    shots: int,
+    correct_bitstring: str,
+    is_noisy: bool = True,
+) -> List[float]:
+    """Executes the input circuit(s) and returns ⟨A⟩, where
+    A = |correct_bitstring⟩⟨correct_bitstring| for each circuit.
+    """
+
+    # Compile circuits to appropriate gateset.
+    if is_noisy:
+        circuit_to_run = circuit.with_noise(cirq.amplitude_damp(0.05))
+    else:
+        circuit_to_run = circuit.with_noise(cirq.amplitude_damp(0.05))
+
+    backend = cirq.Simulator()
+
+    counts = 0
+    for _ in range(shots):
+        result = backend.run(
+            circuit_to_run + cirq.measure(*sorted(circuit.all_qubits()))
+        )
+        bitstring = list(result.measurements.values())[0][0].tolist()
+        if bitstring == correct_bitstring:
+            counts += 1
```

Comment:
`correct_bitstring` is typehinted to be a string, but it look like bitstring is a list. Which is correct?

Reply:
Ah! typehint should be `List[int]`